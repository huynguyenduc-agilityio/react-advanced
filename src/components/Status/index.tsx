import { Flex, Text } from '@chakra-ui/react';

// Icons
import { EllipseIcon } from '@/icons';

// Enums
import { Status as StatusEnum } from '@/enums';

export interface StatusProps {
  type?: StatusEnum;
}

const Status = ({ type = StatusEnum.Online }: StatusProps) => {
  return (
    <Flex
      maxW="60px"
      alignItems="center"
      justifyContent="center"
      gap={1}
      px={2}
      py={1}
      borderRadius="2px"
      border="1px solid"
      bg={type === StatusEnum.Online ? '#09363B' : '#39425F'}
      borderColor={type === StatusEnum.Online ? '#067B51' : '#515A79'}
    >
      <EllipseIcon
        color={(type !== StatusEnum.Online && '#AEB9E1') || undefined}
      />
      <Text
        fontWeight="medium"
        fontSize="xs"
        color={type === StatusEnum.Online ? '#14CA74' : 'pastelBlue'}
      >
        {type === StatusEnum.Online ? 'Online' : 'Offline'}
      </Text>
    </Flex>
  );
};

export default Status;
