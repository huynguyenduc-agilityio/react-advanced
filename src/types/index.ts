export * from './user';
export * from './router';
export * from './pagination';
export * from './queryKey';
export * from './httpsRequest';
export * from './table';
export * from './status';
export * from './type';
