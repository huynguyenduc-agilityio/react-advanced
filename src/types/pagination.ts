export interface PaginationTableType {
  currentPage: number;
  totalPage: number;
}

export interface TPagination<T> {
  meta: {
    pagination: {
      page: number;
      pageSize: number;
      pageCount: number;
      total: number;
    };
  };
  items: T[];
}
