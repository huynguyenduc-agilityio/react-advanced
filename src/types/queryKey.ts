export type TQueryKey = {
  page: number;
  pageSize: number;
  filter?: Record<string, string>;
};
