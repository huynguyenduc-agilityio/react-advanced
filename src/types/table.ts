import { ElementType, JSX } from 'react';

export type TDataSource = {
  id: string;
  [key: string]: string | string[] | number | boolean | undefined;
};

export type THeaderTable = {
  title?: string;
  key?: string;
  icon?: ElementType;
  isSort?: boolean;
  renderBody?: (_: TDataSource, index?: number) => JSX.Element;
};

export type TCellTable = {
  key?: string;
  content?: string | number | boolean | JSX.Element | string[] | undefined;
};
