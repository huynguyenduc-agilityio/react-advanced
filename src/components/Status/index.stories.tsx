import type { Meta, StoryObj } from '@storybook/react';

// Types
import { Status as StatusEnum } from '@/types';

// Components
import Status from '.';

const meta = {
  title: 'Components/Status',
  component: Status,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
} satisfies Meta<typeof Status>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Online: Story = {
  args: {
    type: StatusEnum.Online,
  },
};

export const Offline: Story = {
  args: {
    type: StatusEnum.Offline,
  },
};
