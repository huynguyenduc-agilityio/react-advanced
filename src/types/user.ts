import { NotificationType } from '@/types';

export interface IUserPersonalInfo {
  name: string;
  avatar?: string;
  email: string;
  description?: string;
  phone: string;
  position: string;
  location: string;
  website: string;
  company: string;
  status?: string;
}

export interface IUserTeamInfo {
  teamName: string;
  rank: string;
  office: string;
  teamMail: string;
}

export interface IUserBillInfo {
  payment: string;
  billName: string;
  billAddress: string;
  state: string;
  zipCode: string;
}

export interface IUserNotifications {
  mentionMessage?: NotificationType;
  replyMessage?: NotificationType;
  assignTask?: NotificationType;
  taskOverdue?: NotificationType;
  dailySummary?: NotificationType;
  weeklySummary?: NotificationType;
  monthlySummary?: NotificationType;
  annuallySummary?: NotificationType;
}

export interface IUserModel
  extends IUserPersonalInfo,
    IUserTeamInfo,
    IUserBillInfo,
    IUserNotifications {
  id?: string;
}

export interface IUsers {
  meta: {
    pagination: {
      page: number;
      pageSize: number;
      pageCount: number;
      total: number;
    };
  };
  items: IUserModel[];
}
