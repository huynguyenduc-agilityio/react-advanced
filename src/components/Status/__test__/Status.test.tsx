import { render, screen } from '@testing-library/react';

// Types
import { Status as StatusEnum } from '@/types';

// Component
import Status from '@/components/Status';

describe('Status Component', () => {
  it('renders with default status (Online)', () => {
    render(<Status />);

    expect(screen.getByText('Online')).toBeInTheDocument();
  });

  it('renders with Offline status', () => {
    render(<Status type={StatusEnum.Offline} />);

    expect(screen.getByText('Offline')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<Status />);

    expect(container).toMatchSnapshot();
  });
});
