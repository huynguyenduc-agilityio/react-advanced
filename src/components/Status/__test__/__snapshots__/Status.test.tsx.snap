// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status Component should match snapshot 1`] = `
<div>
  <div
    class="css-1i6zq7o"
  >
    <svg
      fill="none"
      height="4"
      viewBox="0 0 4 4"
      width="4"
    >
      <circle
        cx="2.4375"
        cy="1.59375"
        fill="#05C168"
        r="1.5"
      />
    </svg>
    <p
      class="chakra-text css-1k5z805"
    >
      Online
    </p>
  </div>
</div>
`;
